Tools/Scripts/layout-test-failure-analysis

```diff
@@ -0,0 +1,291 @@
+
+        # export each worker separately
+        farr = filename.split(".")
+        for n in data.keys():
+            ftmparr = farr[:]
+            ftmparr[0] += str(n)
+            worker = data.get(n)
+            export_file(f'{dir}/{".".join(ftmparr)}', '\n'.join(worker), no_warn)
+        info(f'Finished exporting worker test run logs to "{dir}".')
+
+
+# Parses a raw stdio log into a str containing a readable list of tests.
+# Used when exporting a combined worker list.
+def parse_raw_stdio_into_test_names(raw, target_test='', long=False):
+    info("Parsing into chronological list of tests...")
+    data = raw.split("\n")
+    tests = []
+    i = 0
+    for line in data:
+        m = re.findall(r"worker\/(\d+) ([^\"\s].*\.(htm|html|pl|py|svg|xht|xhtml|xml)([?]?.*)) (passed|failed)", line)
+        if len(m) > 0:
+            # continue
+            i += 1
+            test_name = m[0][1]
+            tests.append(test_name)
+            if not long and test_name == target_test:
+                info(f'Found requested test (last of {i}), skipping remaining tests.')
+                break
+    print(f'Processed {i} tests total.')
+    return "\n".join(tests)
+
+
+# Parses a raw stdio log into a dict of lists containing test lists.
+# Used when exporting one worker, or all as separate files.
+def parse_raw_stdio_into_workers(raw, target_test='', long=False):
+    info("Parsing into workers...")
+    data = raw.split("\n")
+    workers = {}
+    target_worker = -1
+    i = 0
+    for line in data:
+        m = re.findall(r"worker\/(\d+) ([^\"\s]+.+.(htm|html|pl|py|svg|xht|xhtml|xml)([?]?.*)) (passed|failed)", line)
+        if len(m) > 0:
+            worker, test_name = (str(m[0][0]), m[0][1])
+            i += 1
+            if type(workers.get(worker)) != list:
+                workers.update({f'{worker}': []})
+            workers[worker].append(test_name)
+            if test_name == target_test:
+                info(f'Found requested test in worker {worker} (test run #{len(workers[worker])} in that worker).')
+                target_worker = worker
+                if long:
+                    info(f'Continuing to sort tests by worker (--long).')
+                    continue
+                break
+    print(f'Finished parsing tests ({i} total).')
+    return (workers, int(target_worker))
+
+
+# Loads the stdio input for analysis from a file.
+# Returns a str with the contents of the stdio log.
+def load_from_file(path):
+    info("Loading data from file...")
+    if not os.path.isfile(path):
+        error(f'Could not find a file at the specified path "{path}".')
+        graceful_exit(os.EX_DATAERR)
+    text = ""
+    with open(path) as file:
+        for line in file:
+            text += line
+    return text
+
+
+# Loads the stdio input for analysis from a build URL.
+# Returns a str with the contents of the stdio log.
+def load_from_url(url):
```

Comment:
Specifically, you're loading from a buildbot URL. Which is probably important to specify in this function name.